core/src/main/java/org/frankframework/pipes/DataSonnetPipe.java

```diff
@@ -208,8 +217,11 @@ private FrankMessageDocument(Message message, MediaType mediaType) {
 			}
 		}
 
-		private static MediaType getMimeType(Message message) {
+		private static MediaType getMimeType(Message message, boolean computeMimeType) {
 			MimeType springMime = message.getContext().getMimeType();
+			if(springMime == null && computeMimeType) {
+				springMime = MessageUtils.computeMimeType(message);
+			}
```

Comment:
Kan dit naar een aparte method? Bv `getMimeType` (dan kan deze method hernoemd worden naar `getMediaType`). Dan heb je hier niet multiple reassignment aan local var springMime. 

`MimeType springMime = getMimeType(message, computeMimeType);`

Reply:
Ik weet niet hoe netjes dat is aangezien deze methode aangeroepen wordt door de constructor. maar ik zal is kijken..

Reply:
Wellicht heb ik niet de volledige context. Als het teveel gedoe is, dan niet ;-)